### PR TITLE
docs(framework) Update save model tutorial

### PR DESCRIPTION
**Pull request by yan-gao-GY**
## Proposal

Update save/load model tutorial: https://flower.ai/docs/framework/how-to-save-and-load-model-checkpoints.html

### CODE REVIEW

File: doc/source/how-to-save-and-load-model-checkpoints.rst [hunk 1]
```diff
@@ -21,35 +21,34 @@ returns those aggregated weights to the caller (i.e., the server):
...
-                # Save aggregated_ndarrays
+                # Save aggregated_ndarrays to disk
                 print(f"Saving round {server_round} aggregated_ndarrays...")
                 np.savez(f"round-{server_round}-weights.npz", *aggregated_ndarrays)
 
             return aggregated_parameters, aggregated_metrics
 
 
-    # Create strategy and run server
+    # Create strategy and pass into ServerApp
```
Comment (chongshenng) [inline doc/source/how-to-save-and-load-model-checkpoints.rst:46]: Let's add the structure of the `server_fn` here for clarity:
``` python
    def server_fn(context):
        strategy = SaveModelStrategy(
            # (same arguments as FedAvg here)
        )
        config = ServerConfig(
            # (your server configs here, like `num-server-rounds`)
        )
        return ServerAppComponents(strategy=strategy, config=config)```

**Review by chongshenng — APPROVED** [review]
lgtm!

File: doc/source/how-to-save-and-load-model-checkpoints.rst [hunk 2]
```diff
@@ -21,35 +21,42 @@ returns those aggregated weights to the caller (i.e., the server):
...
-    )
-    fl.server.start_server(strategy=strategy)
+    # Create strategy and pass into ServerApp
+    def server_fn(context):
+        strategy = SaveModelStrategy(
+            # (same arguments as FedAvg here)
+        )
+        config = ServerConfig(
+            # (your server configs here, like `num-server-rounds`)
+        )
```
Comment (jafermarq) [inline doc/source/how-to-save-and-load-model-checkpoints.rst:53]:
```suggestion
        config = ServerConfig(num_rounds=3)
```

Comment (jafermarq) [reply doc/source/how-to-save-and-load-model-checkpoints.rst:53]: `num-server-rounds` is not a key in `ServerConfig`. I'm thinking showing it like i propose above keeps things simpler and easier for others to copy right away and use or modify.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**